src/neo/Ledger/Blockchain.cs

```diff
@@ -72,7 +73,10 @@ private class UnverifiedBlocksList { public LinkedList<Block> Blocks = new Linke
         public DataCache View => new SnapshotCache(Store);
         public MemoryPool MemPool { get; }
         public uint Height => NativeContract.Ledger.CurrentIndex(currentSnapshot);
+        public uint HeaderHeight => header_cache.Count > 0 ? header_cache[^1].Index : NativeContract.Ledger.CurrentIndex(currentSnapshot);
         public UInt256 CurrentBlockHash => NativeContract.Ledger.CurrentHash(currentSnapshot);
+        public UInt256 CurrentHeaderHash => header_cache.Count > 0 ? header_cache[^1].Hash : NativeContract.Ledger.CurrentHash(currentSnapshot);
```

Comment:
@Qiao-Jin Please resolve the conflicts and remove these properties.

Reply:
`HeadHeight` and `CurrentHeaderHash` is needed for different actors. Should we add a read/write lock in header_cache?

Reply:
For example, `Taskmanager` needs to know the biggest index in `Blockchain`'s header cache so that it can decide whether to send `GetHeaders` or `GetBlocks`.